webapp/src/components/App/Singlestudy/explore/Configuration/General/dialogs/ScenarioBuilderDialog/utils.ts

```diff
@@ -2,35 +2,231 @@ import { AxiosResponse } from "axios";
 import { StudyMetadata } from "../../../../../../../../common/types";
 import client from "../../../../../../../../services/api/client";
 
-export const TABS_DATA: Array<[string, string]> = [
-  ["load", "l"],
-  ["thermal", "t"],
-  ["hydro", "h"],
-  ["wind", "w"],
-  ["solar", "s"],
-  ["ntc", "ntc"],
-  ["hydroLevels", "hl"],
-];
+////////////////////////////////////////////////////////////////
+// Constants
+////////////////////////////////////////////////////////////////
 
-export const ACTIVE_SCENARIO_PATH =
-  "settings/generaldata/general/active-rules-scenario";
+export const SCENARIOS = [
+  "load",
+  "thermal",
+  "hydro",
+  "wind",
+  "solar",
+  "ntc",
+  "renewable",
+  "bindingConstraints",
+] as const;
 
-export type ScenarioBuilderConfig = Record<string, unknown>;
+export type ScenarioType = (typeof SCENARIOS)[number];
 
-function makeRequestURL(studyId: StudyMetadata["id"]): string {
-  return `v1/studies/${studyId}/config/scenariobuilder`;
+////////////////////////////////////////////////////////////////
+// Types
+////////////////////////////////////////////////////////////////
+
+/**
+ * Represents yearly configuration values, which can be either a numerical value or an uninitialized value represented as an empty string.
+ *
+ * @example
+ * { "0": 120, "1": "", "2": 150 }
+ */
+export type YearlyValues = number | "";
+
+/**
+ * Maps area identifiers to their configuration, each configuration being a series of values or uninitialized states over multiple time periods.
```

Comment:
"over multiple MC years"?

Reply:
some comments are unclear I will simplify this